src/Hangfire.PostgreSql/PostgreSqlDistributedLock.cs

```diff
@@ -173,26 +181,38 @@ public static void Lock(string resource, TimeSpan timeout, IDbConnection connect
           }
         }
 
-        throw new PostgreSqlDistributedLockException($@"Could not place a lock on the resource '{resource}': Lock timeout.");
+        throw new PostgreSqlDistributedLockException($@"Could not place a lock on the resource '{resource}': Lock timeout.", lastException);
       }
 
       private static void TryRemoveLock(string resource, IDbConnection connection, PostgreSqlStorageOptions options)
       {
+        IDbTransaction trx = null;
         try
         {
-          using IDbTransaction transaction = connection.BeginTransaction(IsolationLevel.RepeatableRead);
+          TryBeginTransaction(connection, out trx);
           connection.Execute($@"DELETE FROM ""{options.SchemaName}"".""lock"" WHERE ""resource"" = @Resource AND ""acquired"" < @Timeout",
             new {
               Resource = resource,
               Timeout = DateTime.UtcNow - options.DistributedLockTimeout,
-            }, transaction: transaction);
+            }, trx);
 
-          transaction.Commit();
+          trx?.Commit();
         }
         catch (Exception ex)
         {
           Log(resource, "Failed to remove lock", ex);
         }
+        finally
+        {
+          trx?.Dispose();
+        }
+      }
+
+      private static void TryBeginTransaction(IDbConnection connection, out IDbTransaction trx)
```

Comment:
Developers and naming stuff. Yeah, that's the hardest part.

Renamed to just `BeginTransactionIfNotPresent` with just a return value instead of `out`.